### PR TITLE
Migrating firebase v8 to v9

### PR DESCRIPTION
**Pull request by shivareddy6**
This pr deals with migrating the codebase from firebase v8 to v9

## Description
We currently use firebase v8 and in this pr, I'm planning to update it to firebase v9

## Related Issue
https://github.com/scorelab/Codelabz/issues/713

## Motivation and Context

- Firebase v9 has come up with the modular API interface that allows facilitate tree-shaking (removal of unused code) to make the app as small and fast as possible.
- Some modules don't yet support firebase v9 like react-redux-firebase. So planning to use modular API where ever it is supported and use compat API which is fully compatible with the version 8 of firebase with some syntax changes.

## Changes to be made:

- [x] Change firebase version to 9 (9.17.2)
- [x] Use compat version of firebase v9
- [ ] Use modular API interface where ever is supported.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by Sarfraz-droid** [comment]
@shivareddy6 some tests are failing, can you look into it?

**Comment by shivareddy6** [comment]
> @shivareddy6 some tests are failing, can you look into it?

Working on them

File: src/components/AuthPage/ForgotPassword/index.jsx [hunk 1]
```diff
@@ -43,7 +43,13 @@ const ForgotPassword = ({
 	const loadingProps = useSelector(({ auth }) => auth.profile.loading);
 	const dispatch = useDispatch();
 
-	useEffect(() => setError(errorProps), [errorProps]);
+	useEffect(() => {
+		if (errorProps === "Firebase: Error (auth/missing-email).") {
```
Comment (Sarfraz-droid) [inline src/components/AuthPage/ForgotPassword/index.jsx:47]: @shivareddy6 add the error case at `src/helpers/errorMsgHandler.jsx` instead of here

**Review by Sarfraz-droid — CHANGES_REQUESTED** [review]

File: src/components/AuthPage/ForgotPassword/index.jsx (hunk 1 above)
Comment (shivareddy6) [reply src/components/AuthPage/ForgotPassword/index.jsx:47]: sure

**Comment by shivareddy6** [comment]
@Sarfraz-droid, the reason for failing test cases is the re-rendering of user-dashboard and this is due the errors in "public\firebase-messaging-sw.js". I've researched about it and we need to use firebase config to initialize the app and this error gets resolved.
After initializing the firebase app in "public\firebase-messaging-sw.js":
image

If i dont initialize the app then one test case fails:
image

But the problem is we cant use env variables in public to initialize the app. So for me to pass all the cases, i need to use config keys directly in the public folder which is not advisable. What do you want me to do now?

**Comment by Sarfraz-droid** [comment]
Thankyou @shivareddy6 for pointing this out! I'll too look into the changes and make the needed changes to your repository. for the tests to pass

**Comment by Sarfraz-droid** [comment]
> @Sarfraz-droid, the reason for failing test cases is the re-rendering of user-dashboard and this is due the errors in "public\firebase-messaging-sw.js". I've researched about it and we need to use firebase config to initialize the app and this error gets resolved. After initializing the firebase app in "public\firebase-messaging-sw.js": image
> 
> If i dont initialize the app then one test case fails: image
> 
> But the problem is we cant use env variables in public to initialize the app. So for me to pass all the cases, i need to use config keys directly in the public folder which is not advisable. What do you want me to do now?

@shivareddy6 we can use the env variables in public. As these variables have already been exposed in the app during initialization. We would have to restrict the access using firebase security rules.

**Comment by shivareddy6** [comment]
> Thankyou @shivareddy6 for pointing this out! I'll too look into the changes and make the needed changes to your repository. for the tests to pass

Thanks for the changes @Sarfraz-droid, now the tests are getting passed in my forked repo.

**Pull request merged**